source/opt/desc_sroa.cpp

```diff
@@ -114,7 +117,13 @@ bool DescriptorScalarReplacement::ReplaceCandidate(Instruction* var) {
         switch (use->opcode()) {
           case SpvOpAccessChain:
           case SpvOpInBoundsAccessChain:
-            work_list.push_back(use);
+            access_chain_work_list.push_back(use);
+            return true;
+          case SpvOpLoad:
+            load_work_list.push_back(use);
+            return true;
+          case SpvOpCompositeExtract:
+            extract_work_list.push_back(use);
```

Comment:
Can this happen?  The value would have to be loaded first, and then extracted right?

Reply:
oh yes that's true